akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -118,33 +118,33 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
 
   // Called by the outgoing stream multiplexer when that side of the stream is ended.
   def handleOutgoingEnded(streamId: Int): Unit =
-    updateState(streamId, _.handleOutgoingEnded())
+    updateState(streamId, _.handleOutgoingEnded(), "handleOutgoingEnded")
 
   def handleOutgoingFailed(streamId: Int, cause: Throwable): Unit =
-    updateState(streamId, _.handleOutgoingFailed(cause))
+    updateState(streamId, _.handleOutgoingFailed(cause), "handleOutgoingFailed")
 
   /** Called by multiplexer to distribute changes from INITIAL_WINDOW_SIZE to all streams */
   def distributeWindowDeltaToAllStreams(delta: Int): Unit =
-    updateAllStates {
+    updateAllStates({
       case s: Sending => s.increaseWindow(delta)
       case x          => x
-    }
+    }, "distributeWindowDeltaToAllStreams")
 
   /** Called by the multiplexer if ready to send a data frame */
   def pullNextFrame(streamId: Int, maxSize: Int): PullFrameResult =
-    updateStateAndReturn(streamId, _.pullNextFrame(maxSize))
+    updateStateAndReturn(streamId, _.pullNextFrame(maxSize), "pullNextFrame")
 
   /** Entry-point to handle IncomingStreamBuffer.onPull through the state machine */
   def incomingStreamPulled(streamId: Int): Unit =
-    updateState(streamId, _.incomingStreamPulled())
+    updateState(streamId, _.incomingStreamPulled(), "incomingStreamPulled")
 
-  private def updateAllStates(handle: StreamState => StreamState): Unit =
-    streamStates.keys.foreach(updateState(_, handle))
+  private def updateAllStates(handle: StreamState => StreamState, event: String, eventArg: AnyRef = null): Unit =
```

Comment:
Trying to do this allocation free, required adding `eventArg` even if that info is only relevant for `handleStreamEvent`.